src/camelot/fetchNitroPool.ts

```diff
@@ -0,0 +1,139 @@
+import { BigNumber } from 'ethers'
+
+import { CamelotNitroPool, ERC20 } from '../typechained'
+import { Geb } from '../geb'
+import { fromBigNumber, multicall, CamelotMulticallRequest, SECONDS_IN_YEAR } from '../utils'
+
+export type NitroPoolDetails = {
+    tvl: number
+    pendingRewards: {
+        pending1: number
+        pending2: number
+    }
+    settings: {
+        startTime: BigNumber
+        endTime: BigNumber
+        harvestStartTime: BigNumber
+        depositEndTime: BigNumber
+        lockDurationReq: BigNumber
+        lockEndReq: BigNumber
+        depositAmountReq: BigNumber
+        whitelist: boolean
+        description: string
+    }
+    rewardsPerSecond: number
+    lpTokenBalance: number
+    userInfo: {
+        totalDepositAmount: BigNumber
+        rewardDebtToken1: BigNumber
+        rewardDebtToken2: BigNumber
+        pendingRewardsToken1: BigNumber
+        pendingRewardsToken2: BigNumber
+    } | null
+    nitroRewards1: any
+    nitroRewards2: any
+    apy: number
+}
+
+const fetchNitroPool = async (
+    geb: Geb,
+    poolAddress: string,
+    userAddress: string,
+    tokenAddress: string,
+    collateralAddress: string
+): Promise<NitroPoolDetails> => {
+    const token = geb.getErc20Contract(tokenAddress)
+    const collateral = geb.getErc20Contract(collateralAddress)
+    const camelotNitroPool = geb.getCamelotContract(poolAddress)
+
+    // @to-do: move out and replace with actual market price once pool is deployed
+    // const [odgPrice, odgPriceValidity] = await chainlinkRelayerContractODG.getResultWithValidity()
+    // const [collateralPrice, collateralPriceValidity] = await chainlinkRelayerContract.getResultWithValidity()
+    // const odgMarketPriceFloat = parseFloat(ethers.utils.formatEther(odgPrice))
+    // const collateralPriceFloat = parseFloat(ethers.utils.formatEther(collateralPrice))
+
+    const results = await Promise.all([
```

Comment:
Add sequential call to fetch nftPool -> lpToken -> lp deposit collateral tokens (eg. OD, ETH)